.github/container/Dockerfile.jax

```diff
@@ -47,48 +54,63 @@ RUN build-jax.sh \
     --xla-arm64-patch /opt/xla-arm64-neon.patch \ 
     --clean
 
-RUN cp -r ${SRC_PATH_JAX} ${SRC_PATH_JAX}-no-git && rm -rf ${SRC_PATH_JAX}-no-git/.git
-RUN cp -r ${SRC_PATH_XLA} ${SRC_PATH_XLA}-no-git && rm -rf ${SRC_PATH_XLA}-no-git/.git
-
 ###############################################################################
-## Build 'runtime' flavor without the git metadata
+## Pack jaxlib wheel and various source dirs into a pre-installation image
 ###############################################################################
 
 ARG BASE_IMAGE
-FROM ${BASE_IMAGE} as runtime-image
+FROM ${BASE_IMAGE} as mealkit
+ARG SRC_MANIFEST_FILE
+ARG DEST_MANIFEST_DIR
 ARG SRC_PATH_JAX
 ARG SRC_PATH_XLA
+ARG SRC_PATH_TE
 ARG BUILD_DATE
+
 ENV BUILD_DATE=${BUILD_DATE}
 # The following environment variables tune performance
 ENV XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=true --xla_gpu_enable_async_all_gather=true --xla_gpu_enable_async_reduce_scatter=true --xla_gpu_enable_triton_gemm=false"
 ENV CUDA_DEVICE_MAX_CONNECTIONS=1
 ENV NCCL_IB_SL=1
 ENV CUDA_MODULE_LOADING=EAGER
+# Set the manifest env vars
+ENV MANIFEST_FILE=${DEST_MANIFEST_DIR}/${SRC_MANIFEST_FILE}
+
+ADD --chmod=777 create-distribution.sh ${DEST_MANIFEST_DIR}/
+COPY ${SRC_MANIFEST_FILE} ${DEST_MANIFEST_DIR}/${SRC_MANIFEST_FILE}
```

Comment:
Can you use env variable `MANIFEST_FILE` in dockerfile commands?

Reply:
I separated the `MANIFEST_FILE` into the filename itself and the dir, b/c I wanted to be able to use DEST_MANIFEST_DIR at build time for `ADD`/`COPY`, and re-use `SRC_MANIFEST_FILE` at build time to specify what "manifest file" to pull from the context, but also what the file should be named in `DEST_MANIFEST_DIR`

Do you see any other way to accomplish this? I implemented it this way in case down the road we have one manifest for arm and one for amd.

Reply:
Also, the `DEST_MANIFEST_DIR` is also used in the copy of the patches, and I can't `dirname` the `MANIFEST_FILE` since I need the dirname to be known at build time

Reply:
@DwarKapex and I synced offline on this, because of the limitation of dockerfile's not allowing bash commands at build time, the tradeoff here is between:

## Current
```dockerfile
ARG SRC_MANIFEST_FILE=manifest.yaml
ARG DEST_MANIFEST_DIR=/opt/manifest.d
ENV MANIFEST_FILE=${DEST_MANIFEST_DIR}/${SRC_MANIFEST_FILE}
ADD --chmod=777 create-distribution.sh ${DEST_MANIFEST_DIR}/
COPY ${SRC_MANIFEST_FILE} ${DEST_MANIFEST_DIR}/${SRC_MANIFEST_FILE}
COPY patches/ ${DEST_MANIFEST_DIR}/patches/
```
vs

## Proposed
```dockerfile
ARG MANIFEST_FILE=/opt/manifest.d/manifest.yaml
ARG DEST_MANIFEST_DIR=/opt/manifest.d
ENV MANIFEST_FILE=${MANIFEST_FILE}
ADD --chmod=777 create-distribution.sh ${DEST_MANIFEST_DIR}/
COPY ${SRC_MANIFEST_FILE} ${MANIFEST_FILE}
COPY patches/ ${DEST_MANIFEST_DIR}/patches/
```

Both have the issue that you must repeat something and there's possible for error. In the `Current` state, a developer changing `${DEST_MANIFEST_DIR}/${SRC_MANIFEST_FILE}` needs to do so in two places, which could lead to errors and in `Proposed` we need to make sure the `--build-args` both have the same absolute path prefix, otherwise things won't work. We have decided to stick with `Current` for now since a developer change is less frequent than a build arg change. We can revisit later if there's a more elegant way to accomplish this

### Aside
You can also use a separate build context and mount it into a `RUN` command, but we decided it was too much complexity and to just stick with either of these.